### PR TITLE
test: _transfer internal function

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes 0XB-432

### CODE REVIEW

File: packages/contracts/test/unit/core/BatchRelayer.t.sol [hunk 1]
```diff
@@ -37,15 +37,34 @@ contract PrivacyPoolForTest {
...
 
+contract BatchRelayerForTest is BatchRelayer {
+  constructor(uint256 _maxRelayFeeBPS) BatchRelayer(_maxRelayFeeBPS) {}
+
+  function forTest_transfer(IERC20 _asset, address _recipient, uint256 _amount) external {
+    _transfer(_asset, _recipient, _amount);
+  }
+}
+
+contract ForTest_ReceiveRevert {
```
Comment (0xteddybear) [inline packages/contracts/test/unit/core/BatchRelayer.t.sol]: `ReceiveRevertForTest` ? in order to keep it consistent with the above

File: packages/contracts/test/unit/core/BatchRelayer.tree [hunk 2]
```diff
@@ -14,4 +14,15 @@ UnitBatchRelayer::batchRelay
...
+
+UnitBatchRelayer::_transfer
+├── When recipient is zero
+│   └── It reverts with ZeroAddress
+├── When asset is native
+│   ├── When transfer fails
+│   │   └── It reverts with NativeAssetTransferFailed
+│   └── When transfer succeeds
+│       └── It transfers the amount to the recipient
+└── When asset is not native
```
Comment (0xteddybear) [inline packages/contracts/test/unit/core/BatchRelayer.tree:27]: add error cases for
- the asset returning `false`
- the asset reverting

File: packages/contracts/test/unit/core/BatchRelayer.t.sol [hunk 3]
```diff
@@ -236,4 +255,47 @@ contract UnitBatchRelayer is Test {
...
+    vm.assume(_recipient != address(0));
+    vm.deal(address(batchRelayerForTest), _amount);
+
+    // It transfers the amount to the recipient
+    vm.expectCall(_recipient, _amount, '');
+
+    batchRelayerForTest.forTest_transfer(IERC20(Constants.NATIVE_ASSET), _recipient, _amount);
+  }
+
+  function test__transferWhenAssetIsNotNative(IERC20 _asset, address _recipient, uint256 _amount) external {
```
Comment (0xteddybear) [inline packages/contracts/test/unit/core/BatchRelayer.t.sol:291]: shouldn't you `vm.assume(_asset  != Constants.NATIVE_ASSET)`?

**Review by 0xteddybear — CHANGES_REQUESTED** [review]

File: packages/contracts/test/unit/core/BatchRelayer.tree (hunk 2 above)
Comment (0xAustrian) [reply packages/contracts/test/unit/core/BatchRelayer.tree:27]: Don't you think that is already tested on the tests from `safeTransfer` (OZ test suite)?

File: packages/contracts/test/unit/core/BatchRelayer.t.sol (hunk 3 above)
Comment (0xAustrian) [reply packages/contracts/test/unit/core/BatchRelayer.t.sol:291]: https://github.com/defi-wonderland/privacy-pools-core/pull/8/commits/e2282c4c69e3f45c2c1d6b0cb0a098699fe25a9f

File: packages/contracts/test/unit/core/BatchRelayer.t.sol (hunk 1 above)
Comment (0xAustrian) [reply packages/contracts/test/unit/core/BatchRelayer.t.sol]: https://github.com/defi-wonderland/privacy-pools-core/pull/8/commits/e2282c4c69e3f45c2c1d6b0cb0a098699fe25a9f

File: packages/contracts/test/unit/core/BatchRelayer.tree (hunk 2 above)
Comment (0xteddybear) [reply packages/contracts/test/unit/core/BatchRelayer.tree:27]: Ideally I'd argue for asserting we're actually calling `safeTransfer` via `vm.expectCall`, but since it's a library and not an external call, then we have to resort to testing its behaviour if we want to avoid the possibility of later down the line switching to a bare `.transfer` call and have all tests still pass

I'll admit that's an edge-case mutation to attempt to catch though :grin: feel free to add it or not

**Review by 0xteddybear — APPROVED** [review]
